### ISSUE TITLE
/camera/depth/disparity missing

### ISSUE THREAD

**Issue by ToMadoRe** [strands-project/scitos_robot#46]
I am just trying to get this (https://github.com/strands-project/strands_utils/tree/hydro-devel/driveable_floor_calib) working.

However, I am missing the disparity topic when using the openni_wrapper. Even though the topic /depth/disparity is shown at rostopic list, the qt based calibration tool does not show anything.

**Comment by RaresAmbrus** [comment]
Since no one was using that topic, I haven't implemented it. Do you need it?

**Comment by ToMadoRe** [comment]
Well, yes. I would need it to test and use that module for cliff detection. Is it hard to implement it?

**Comment by RaresAmbrus** [comment]
Nope, it should be pretty easy. I've tried to publish only those things that we actually need. I'll do it today and merge it.

**Comment by ToMadoRe** [comment]
Great, thanks!

**Comment by RaresAmbrus** [comment]
@ToMadoRe for some reason, the disparity nodelet from depth_image_proc (http://wiki.ros.org/depth_image_proc#depth_image_proc.2BAC8-disparity)  doesn't publish the disparity, even though I'm pretty sure I've remapped the topics correctly. And it doesn't print any output either so I can't tell what's wrong. I'll look into it further and let you know once I find a way to fix it.

**Comment by ToMadoRe** [comment]
okay... strange... let me know if you have a solution

**Comment by RaresAmbrus** [comment]
@ToMadoRe after some debugging I've figured the reasons why the disparity nodelet wasn't producing any output: 
-  It's using a synchronizer for the depth image and camera_info streams with an ExactTime policy, which means the two streams need to have exactly the same timestamps (this is not the case for other nodelets which use the ApproximateTime policy). I've taken care of this.
- It assumes the baseline between the two cameras is encoded in the P matrix of the cam_info message (unlike other nodelets which calculate the baseline from the tf). By default the baseline isn't published, but I've added that now, and it seems to work. The disparity is published in `/camera/depth/disparity` (where camera is the namespace you've chosen). 

I've committed the changes to my branch, can you please test that out before I merge it ? I'll open a pull request so you have access to it.

**Comment by cburbridge** [comment]
@RaresAmbrus I think this is done?
